webpack/scenes/ContentViews/Create/__tests__/createContentView.test.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { renderWithRedux, patientlyWaitFor, fireEvent } from 'react-testing-lib-wrapper';
+
+import { nockInstance, assertNockRequest } from '../../../../test-utils/nockWrapper';
+import api from '../../../../services/api';
+import CreateContentViewForm from '../CreateContentViewForm';
+
+const cvCreateData = require('./contentViewCreateResult.fixtures.json');
+
+const cvCreatePath = api.getApiUrl('/content_views');
+
+const setModalOpen = jest.fn();
+
+const createDetails = {
+  name: '1232123',
+  label: '1232123',
+  description: '',
+  composite: false,
+  dependencies: false,
+  auto_publish: false,
+  import_only: false,
+};
+
+const createdCVDetails = { ...cvCreateData };
+
+const form = <CreateContentViewForm setModalOpen={setModalOpen} />;
+
+test('Can save content view from form', async (done) => {
+  const createscope = nockInstance
+    .post(cvCreatePath, createDetails)
+    .reply(201, createdCVDetails);
+  const { queryByText, getByLabelText } = renderWithRedux(form);
+  expect(queryByText('Description')).toBeInTheDocument();
+
+  fireEvent.change(getByLabelText('input_name'), { target: { value: '1232123' } });
+
+  await patientlyWaitFor(() => { expect(getByLabelText('input_label')).toHaveAttribute('value', '1232123'); });
+
+  getByLabelText('create_content_view').click();
+
+  assertNockRequest(createscope, done);
+});
+
+test('Form closes itself upon save', async (done) => {
+  const createscope = nockInstance
+    .post(cvCreatePath, createDetails)
+    .reply(201, createdCVDetails);
+  const { queryByText, getByLabelText } = renderWithRedux(form);
+  expect(queryByText('Description')).toBeInTheDocument();
```

Comment:
A subtle thing with react-testing-library that took me a bit to figure out is when to use `getBy*` and `queryBy*`, I finally landed on:
`getBy*` is for when you expect the element to be present
`queryBy*` is for when you don't expect the element to be present

The big difference is the error message for `getBy*` is more informative since it raises an error when the element is not there, vs. `queryBy*` returns null and you get an error message about not being able to find null

Can you update anything that is expected to be there to `getBy*`?

Reply:
Updated to use getBy* for expected elements and queryBy for elements not expected to be present.
The difference I saw 
getByText('Description')).not.toBeInTheDocument(); was it doesn't fail at _.not.toBeInTheDocument()_ but instead fails at getByText itself if element is not in document.